src/signTransaction.c

```diff
@@ -34,48 +34,37 @@ enum {
 // &ctx->dhContext) are needed for encryption
 static void processShaAndPosibleDHAndPrepareResponse() {
     if (ctx->dhIsActive) {
-        dh_aes_key_t aesKey;
-        BEGIN_TRY {
-            TRY {
-                // Compute AES key
-                dh_init_aes_key(&aesKey, &ctx->wittnessPath, &ctx->otherPubkey);
-
-                // Encode message chunk
-                ctx->responseLength = dh_encode_append(&ctx->dhContext,
-                                                       &aesKey,
-                                                       ctx->dataToAppendToTx,
-                                                       ctx->dataToAppendToTxLen,
-                                                       G_io_apdu_buffer,
-                                                       SIZEOF(G_io_apdu_buffer));
-                sha_256_append(&ctx->hashContext, G_io_apdu_buffer, ctx->responseLength);
-                VALIDATE(
-                    ctx->countedSectionDifference + ctx->responseLength >= ctx->dataToAppendToTxLen,
-                    ERR_INVALID_STATE);
-                ctx->countedSectionDifference =
-                    ctx->countedSectionDifference + ctx->responseLength - ctx->dataToAppendToTxLen;
-                TRACE("CS diff %d from:%d, %d",
-                      (int) ctx->countedSectionDifference,
-                      (int) ctx->responseLength,
-                      (int) ctx->dataToAppendToTxLen);
-            }
-            FINALLY {
-                explicit_bzero(&aesKey, SIZEOF(aesKey));
-            }
+        uint16_t bufferLen = SIZEOF(G_io_apdu_buffer);
+        uint16_t err = dh_encode_append(&ctx->dhContext,
+                                        &ctx->wittnessPath,
+                                        &ctx->otherPubkey,
+                                        ctx->dataToAppendToTx,
+                                        ctx->dataToAppendToTxLen,
+                                        G_io_apdu_buffer,
+                                        &bufferLen);
+        if (err != SUCCESS) {
+            THROW(err);
         }
-        END_TRY;
+        ctx->responseLength = bufferLen;
+        CX_THROW(sha_256_append(&ctx->hashContext, G_io_apdu_buffer, ctx->responseLength));
+        VALIDATE(ctx->countedSectionDifference + ctx->responseLength >= ctx->dataToAppendToTxLen,
+                 ERR_INVALID_STATE);
+        ctx->countedSectionDifference =
+            ctx->countedSectionDifference + ctx->responseLength - ctx->dataToAppendToTxLen;
+        TRACE("CS diff %d from:%d, %d",
+              (int) ctx->countedSectionDifference,
+              (int) ctx->responseLength,
+              (int) ctx->dataToAppendToTxLen);
     } else {
-        sha_256_append(&ctx->hashContext, ctx->dataToAppendToTx, ctx->dataToAppendToTxLen);
+        cx_err_t err =
+            sha_256_append(&ctx->hashContext, ctx->dataToAppendToTx, ctx->dataToAppendToTxLen);
+        ASSERT(err == CX_OK);
```

Comment:
Not sure why this is different from the treatment of errors above (cx_throw vs. assert).